Content.Server/Atmos/EntitySystems/GasTileOverlaySystem.cs

```diff
@@ -120,40 +120,49 @@ private void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs e)
         /// </summary>
         private void UpdateChunkTile(GridAtmosphereComponent gridAtmosphere, GasOverlayChunk chunk, Vector2i index, GameTick curTick)
         {
-            var oldData = chunk.GetData(index);
+            ref var oldData = ref chunk.GetData(index);
             if (!gridAtmosphere.Tiles.TryGetValue(index, out var tile))
             {
-                if (oldData == null)
+                if (oldData.Equals(default))
                     return;
 
                 chunk.LastUpdate = curTick;
-                chunk.SetData(index, null);
+                oldData = default;
                 return;
             }
 
-            var opacity = new byte[VisibleGasId.Length];
-            GasOverlayData newData = new(tile!.Hotspot.State, opacity);
+            var changed = oldData.Equals(default) || oldData.FireState != tile.Hotspot.State;
+            if (oldData.Equals(default))
+                oldData = new GasOverlayData(tile.Hotspot.State, new byte[VisibleGasId.Length]);
+
             if (tile.Air != null)
             {
-                var i = 0;
-                foreach (var id in VisibleGasId)
+                for (var i = 0; i < VisibleGasId.Length; i++)
                 {
+                    var id = VisibleGasId[i];
                     var gas = _atmosphereSystem.GetGas(id);
                     var moles = tile.Air.Moles[id];
 
-                    if (moles >= gas.GasMolesVisible)
-                    {
-                        opacity[i] = (byte) (ContentHelpers.RoundToLevels(
-                            MathHelper.Clamp01((moles - gas.GasMolesVisible) / (gas.GasMolesVisibleMax - gas.GasMolesVisible)) * 255, byte.MaxValue, _thresholds) * 255 / (_thresholds - 1));
-                    }
-                    i++;
+                    if (moles < gas.GasMolesVisible)
+                        continue;
```

Comment:
this now that its re-using old arrays this has to set `oldData.Opacity[i] = 0`

Reply:
Fixed in [`fc01c3a` (#12617)](https://github.com/space-wizards/space-station-14/pull/12617/commits/fc01c3aa9403e4d76b256f96829cbce360b98f31)